### PR TITLE
palindrome-product: Clarify requirements

### PR DESCRIPTION
**Pull request by dkinzer**
REF exercism/ruby#305
REF exercism/ruby#723

It's not clear from the current examples what the requirements of this exercise are.  In fact, a furtive glance is more likely to cause confusion than give clarity.

In my opinion, this is because for the first example the solution lies within the range of the list of factors, giving on a first impression that we are looking for products that lie within this range.

I think that by listing out the entire list of products (which is small in the case of the range of 1..9) we make it more clear that we are considering all the results.

### CODE REVIEW

File: exercises/palindrome-products/description.md [hunk 1]
```diff
@@ -21,8 +21,14 @@ palindromic product of factors within that range, along with all the factors in
...
-The smallest product is `1`. Its factors are `(1, 1)`.
-The largest product is `9`. Its factors are `(1, 9)`, and `(3, 3)`.
+And given the list of all possible products within this range:
+`[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 12, 14, 16, 18, 15, 21, 24, 27, 20, 28, 32, 36, 25, 30, 35, 40, 45, 42, 48, 54, 49, 56, 63, 64, 72, 81]`
+
+The palindrome products are all single digit numbers (in this case):
+`[1, 2, 3, 4, 5, 6, 7, 8, 9]`
+
+The smallest palindrome product is `1`. It's factors are `(1, 1)`.
+The largest palindrome product is `9`. It's factors are `(1, 9)`, `(3, 3)`, and `(9, 1)`.
```
Comment (Insti) [inline exercises/palindrome-products/description.md]: `It's` should be `Its`

**Comment by Insti** [comment]
Looks good 👍 
Thanks :heart:

**Comment by Insti** [comment]
Once this is merged it would be great if you would create an issue or a PR back here about updating the readme from the updated common description.

There is some documentation about how to use `configlet` to generate the readmes:
https://github.com/exercism/docs/blob/master/language-tracks/exercises/anatomy/readmes.md

**Comment by dkinzer** [comment]
Sure, I'll give that a try once this merged.

File: exercises/palindrome-products/description.md [hunk 2]
```diff
@@ -21,8 +21,14 @@ palindromic product of factors within that range, along with all the factors in
...
-The smallest product is `1`. Its factors are `(1, 1)`.
-The largest product is `9`. Its factors are `(1, 9)`, and `(3, 3)`.
+And given the list of all possible products within this range:
+`[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 12, 14, 16, 18, 15, 21, 24, 27, 20, 28, 32, 36, 25, 30, 35, 40, 45, 42, 48, 54, 49, 56, 63, 64, 72, 81]`
+
+The palindrome products are all single digit numbers (in this case):
+`[1, 2, 3, 4, 5, 6, 7, 8, 9]`
+
+The smallest palindrome product is `1`. Its factors are `(1, 1)`.
+The largest palindrome product is `9`. Its factors are `(1, 9)`, `(3, 3)`, and `(9, 1)`.
```
Comment (Insti) [inline exercises/palindrome-products/description.md]: [The tests](https://github.com/exercism/problem-specifications/blob/master/exercises/palindrome-products/canonical-data.json#L23) only expect (1,9) and (3,3) as factors.
(and always seem to expect factor pairs in (lowest, highest) order.

Comment (dkinzer) [reply exercises/palindrome-products/description.md]: Good point... I forgot I had to sort and uniq my results.   I'll udpdate.

**Comment by jpreese** [comment]
Thanks @dkinzer 🎉 !

**Pull request merged**